telemetry/init.go

```diff
@@ -0,0 +1,42 @@
+package telemetry
+
+// Inspired by the example here:
+// https://github.com/open-telemetry/opentelemetry-go/blob/main/example/prometheus/main.go
+
+import (
+	"context"
+
+	"github.com/gnolang/gno/telemetry/metrics"
+	"github.com/gnolang/gno/telemetry/options"
+)
+
+const (
+	defaultMeterName   = "gno.land"
+	defaultServiceName = "gno.land"
+)
+
+var config options.Config
+
+// MetricsEnabled returns true if metrics have been initialized.
+func MetricsEnabled() bool {
+	return config.MetricsEnabled
+}
+
+// Init will initialize metrics with the options provided. This function may also intialize tracing when
+// this is something that we want to support.
+func Init(ctx context.Context, options ...Option) error {
+	config.MeterName = defaultMeterName
+	config.ServiceName = defaultServiceName
```

Comment:
You set the `ServiceName` but then you do not use anywhere. It is important to initialize a MetricResource with the `ServiceName` and other metadata. See upsteram example from official OTEL documentation

```go
func newResource() (*resource.Resource, error) {
	return resource.Merge(resource.Default(),
		resource.NewWithAttributes(semconv.SchemaURL,
			semconv.ServiceName("my-service"),
			semconv.ServiceVersion("0.1.0"),
		))
}

```

Reply:
Good call; I'll fix it up.